igor-web/src/main/groovy/com/netflix/spinnaker/igor/build/BuildController.groovy

```diff
@@ -94,31 +80,22 @@ class BuildController {
     @RequestMapping(value = '/builds/queue/{master}/{item}')
     Object getQueueLocation(@PathVariable String master, @PathVariable int item) {
         def buildService = getBuildService(master)
-        if (buildService.buildServiceProvider() == BuildServiceProvider.JENKINS) {
-            try {
-                return buildService.getQueuedItem(item)
-            } catch (RetrofitError e) {
-                if (e.response?.status == NOT_FOUND.value()) {
-                    throw new NotFoundException("Queued job '${item}' not found for master '${master}'.")
-                }
-                throw e
-            }
-        } else if (buildService.buildServiceProvider() == BuildServiceProvider.TRAVIS) {
-            return buildService.queuedBuild(item)
+        if (buildService instanceof JenkinsService) {
```

Comment:
Maybe worth encapsulating the logic getting a JenkinsService into a `getJenkinsService()` since it's probably used for each API endpoint.

Reply:
I'm not sure I fully understand.  Ideally we'd completely remove the switch statements in the controller by delegating more to the `BuildService` inheritance chain...but for new we're getting the service (regardless of the type) and just checking what type it is.

Reply:
Not sure I understand why we need to do any casting here? Is `queuedBuild(item)` not part of the `BuildService` interface? If not, should we introduce a `QueuingBuildService` that extends so we don't need to cast this around?

Reply:
It's not part of the interface---I did consider having `JenkinsBuildService` and `TravisBuildService` implement a new `QueuingBuildService` interface which I think is probably the right thing here---but tried to focus this PR more on adding actual typing to the controller via `instanceof` and leave removing the crazy if blocks to a future PR as this was getting to be a big PR.

Reply:
I can appreciate that. At least we're all on the same page as far as some next step improvements go. :)